controllers/space/space_controller_test.go

```diff
@@ -687,20 +678,19 @@ func TestDeleteSpace(t *testing.T) {
 				HaveSpacesForCluster("member-1", 0) // space counter is not incremented when `.spec.TargetCluster` is not set
 		})
 
-		t.Run("when deleting space", func(t *testing.T) {
+		t.Run("when nstemplateset is gone, then it should remove the finalizer and the client automatically delete the Space", func(t *testing.T) {
 			// given
 			s := spacetest.NewSpace("delete-space",
 				spacetest.WithoutSpecTargetCluster(),          // targetCluster is not specified in spec ...
 				spacetest.WithStatusTargetCluster("member-1"), // ... but is available in status
-				spacetest.WithDeletionTimestamp())
+				spacetest.WithDeletionTimestamp(),
+				spacetest.WithFinalizer())
 			hostClient := test.NewFakeClient(t, s, basicTier)
-			nstmplSet := nstemplatetsettest.NewNSTemplateSet("delete-space", nstemplatetsettest.WithReadyCondition())
-			member1Client := test.NewFakeClient(t, nstmplSet)
-			member1Client.MockDelete = mockDeleteNSTemplateSet(member1Client.Client)
+			member1Client := test.NewFakeClient(t)
 			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
 			member2 := NewMemberClusterWithTenantRole(t, "member-2", corev1.ConditionTrue)
 			ctrl := newReconciler(hostClient, member1, member2)
-			InitializeCounters(t, NewToolchainStatus(WithMember("member-1", WithSpaceCount(1))))
+			InitializeCounters(t, NewToolchainStatus(WithEmptyMetrics(), WithMember("member-1", WithSpaceCount(1))))
```

Comment:
just curious - was `NewToolchainStatus` failing with some error when called without this new function `WithEmptyMetrics`. 

Basically I'm trying to understand why this new function is now needed ?

Reply:
There was an issue with the initialization of the counters - in these tests, we want to initialize the counter based on the values provided in the ToolchainStatus, but if the ToolchainStatus doesn't contain some metrics, then it doesn't initialize from the ToolchainStatus, but based on all present resources of the given types in the cluster https://github.com/codeready-toolchain/host-operator/blob/34b494234cf66da80e289113fde7d5e1c0b169a5/pkg/counter/cache.go#L236-L242

So, I would need to create it with:
```
	InitializeCounters(t,
			NewToolchainStatus(
				WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
					"1,internal": 1,
				}),
				WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
					string(metrics.Internal): 1,
				}),
				WithMember("member-1", WithSpaceCount(1)),
			))

```  
but I was tool lazy to copy-paste the whole code, so I thought that the simple function will be easier to use.

Reply:
Ok, I see!  So previously this specific case at line 703 was initializing from resources and not from ToolchainStatus, which is the desired behavior.